TicketManagementSystem/src/main/java/com/endava/TicketManagementSystem/service/OrderService.java

```diff
@@ -0,0 +1,71 @@
+package com.endava.TicketManagementSystem.service;
+
+import com.endava.TicketManagementSystem.dto.GetOrderResponseDTO;
+import com.endava.TicketManagementSystem.model.*;
+import com.endava.TicketManagementSystem.repository.*;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+@Service
+public class OrderService {
+
+    @Autowired
+    OrderRepository orderRepository;
+    @Autowired
+    CustomerRepository customerRepository;
+    @Autowired
+    TicketCategoryRepository ticketCategoryRepository;
+    @Autowired
+    EventRepository eventRepository;
+    @Autowired
+    VenueRepository venueRepository;
+
+    public List<Order> getAllOrders(){
+        return orderRepository.findAll();
+    }
+
+    public List<GetOrderResponseDTO> getOrdersByCustomerId(Integer customerId){
+        List<Order> orders = orderRepository.findOrdersByCustomerId(customerId);
+        List<GetOrderResponseDTO> orderResponses = new ArrayList<>();
+        for (Order order : orders){
+            Integer eventId = ticketCategoryRepository.findAllById(order.getTicketCategory().getId()).get(0).getEvent().getId();
```

Comment:
Here we should have added some conditions but because you had to do a little application and your DB is already populated with valid data... it's fine.
However it's good to know in the future :D , i'll give you an example: if order.getTicketCategory() is null, the following order.getTicketCategory().getId() will throw a NPE(null pointer exception). It is a valid explanation for all the cases from hre.